### PR TITLE
build(pip): Enable the new 2020 resolver

### PR DESCRIPTION
**Pull request by BYK**
This reverts commit c9f3294e9e65787a75538994353141849b198070 and starts using `pip` `20.3.0` which has the new, strict resolver turned on by default.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -79,6 +79,7 @@ RUN set -x \
   " \
   && apt-get update \
   && apt-get install -y --no-install-recommends $buildDeps \
+  && python -m pip install --upgrade 'pip>=20.3.0' \
```
Comment (armenzg) [inline docker/Dockerfile:82]: If we use exact pinning we would have an idempotent step. Would that be preferred?

**Review by armenzg — APPROVED** [review]
There might be more places where this could be upgrade but we can handle one by one.

File: docker/Dockerfile (hunk 1 above)
Comment (BYK) [reply docker/Dockerfile:82]: I'm hoping to remove this step once the base `python-3.6` images start using the latest PIP. If that does not happen, I'll convert this to an exact one.

**Comment by BYK** [comment]
>There might be more places where this could be upgrade but we can handle one by one.

This PR is focusing on our overall compatibility (to ensure that) and the self-hosted releases. Other places fall onto the DevX and Ops teams I think.

**Pull request merged**